crates/bevy_tasks/src/single_threaded_task_pool.rs

```diff
@@ -202,7 +202,7 @@ impl FakeTask {
 /// For more information, see [`TaskPool::scope`].
 #[derive(Debug)]
 pub struct Scope<'scope, 'env: 'scope, T> {
-    executor: &'env async_executor::LocalExecutor<'env>,
```

Comment:
why did you need to change these lifetimes?

Reply:
The futures passed to `scope.spawn` may run the executor and spawn more futures, so they need `&scope`.  That only lives for `'scope`, which is shorter than `'env`.  

The change to this file makes the API match the one in `task_pool.rs`.  

(The difference between the API in `task_pool` and `single_threaded_task_pool` led to some fun and confusing compiler errors.  I somehow had rust-analyzer using a configuration with `task_pool` but `cargo build` using a configuration with `single_threaded_task_pool`, so it worked in the IDE but failed on the command line.  And the lifetime errors it reports don't point at code in `single_threaded_task_pool`, so it took me a while to figure out that this file even existed.)

Reply:
Might be worth pulling out of this pr then. This is a riskier change and if for some reason it gets reverted we should still keep the changes to this file.